pyEpiabm/pyEpiabm/sweep/travel_sweep.py

```diff
@@ -0,0 +1,239 @@
+            HostProgressionSweep.set_infectiousness(person, time)
+            HostProgressionSweep().update_time_status_change(person, time)
+
+    def assign_microcell_household(self, number_individuals_introduced):
+        """
+        Assign introduced individuals to microcells based on population
+        density of micorcells. Take a number of microcells equal to the
+        the number_individuals_introduced (or max number of microcells)
+        and assign individuals randomly to one of these microcells,
+        such that individuals can end up in the same microcell.
+
+        Parameters
+        ----------
+        number_individuals_introduced: int
+            Infected individuals added to population at certain time step
+        """
+        num_microcells_population = sum(len(cell.microcells) for cell in
+                                        self._population.cells)
+        microcells_to_choose_dict = {}
+        for i in range(min(number_individuals_introduced,
+                           num_microcells_population)):
+            # initialise with default microcell
+            microcells_to_choose_dict[Microcell(self.initial_cell)] = 0
+        # Find highest density microcells
+        for possible_cell in self._population.cells:
+            for possible_microcell in possible_cell.microcells:
+                density_microcells_list = list(
+                    microcells_to_choose_dict.values())
+                if min(density_microcells_list) < len(
+                        possible_microcell.persons):
+                    # remove old object
+                    microcells_to_choose_dict.pop(list(
+                        microcells_to_choose_dict.keys())[
+                        list(microcells_to_choose_dict.values()).index(
+                            min(density_microcells_list))])
+                    # add new microcell
+                    microcells_to_choose_dict[possible_microcell] = \
+                        len(possible_microcell.persons)
+
+        # Assign to microcell and household in existing population
+        for person in self.initial_cell.persons:
+            selected_microcell = random.choice(list(
+                microcells_to_choose_dict.keys()))
+            # Assign person to microcell and microcell to person
+            selected_microcell.add_person(person)
+            person.microcell = selected_microcell
+            r = random.random()
+            if r < self.travel_params['prob_existing_household']:
+                # Assign to existing household
+                selected_household = random.choice(
+                    selected_microcell.households)
+                selected_household.add_person(person)
+            else:
+                # Create new household
+                selected_microcell.add_household([person])
+
+    def check_leaving_individuals(self, time, person):
+        """
+        Check if individuals travel_end_time is reached
```

Comment:
How do we calculate `travel_end_time` - this is much more complicated than I was expecting for such a simple description!

Reply:
I added some comments on why we need to check their quarantine/isolation status before removing them.

Reply:
Travel_end_time is just set (time + random number between 2-14 days). Because we are looping over whole population we need to check if it is assigned to a person and if the time > travel_end_time. But we don't want to remove them if they are in isolation/quarantine due to interventions applied to population.

Reply:
OKay that makes sense!